sky/client/sdk.py

```diff
@@ -285,13 +290,20 @@ def optimize(
             for a task.
         exceptions.NoCloudAccessError: if no public clouds are enabled.
     """
-    dag_str = dag_utils.dump_chain_dag_to_yaml_str(dag)
-
-    body = payloads.OptimizeBody(dag=dag_str,
-                                 minimize=minimize,
+    body = payloads.OptimizeBody(minimize=minimize,
                                  request_options=admin_policy_request_options)
+    if dag_id is not None:
+        assert dag is None, 'dag must be None if dag_id is provided'
+        body.dag_id = dag_id
+    else:
+        assert dag is not None, 'dag must be provided if dag_id is not provided'
+        dag_str = dag_utils.dump_chain_dag_to_yaml_str(dag)
+        body.dag = dag_str
+
+    # For backward compatibility, we exclude unset values.
     response = requests.post(f'{server_common.get_server_url()}/optimize',
-                             json=json.loads(body.model_dump_json()),
+                             json=json.loads(
+                                 body.model_dump_json(exclude_unset=True)),
```

Comment:
Should we add this `exclude_unset` everywhere by default? It seems like we need it for backwards compatibility